### PR TITLE
Lex upgrade

### PR DESCRIPTION
**Pull request by astradzhao**
Updates lexical search to parse words as tokens and stems them.
Note that this requires 2 outside libraries.

### CODE REVIEW

File: src/search/search_dictionary.go [hunk 1]
```diff
@@ -12,18 +14,43 @@ import (
...
 	ile.WordOccurrences = make(map[string]int32)
-	tokens := strings.Fields(strings.ToLower(string(content)))
-	for _, token := range tokens {
-		ile.WordOccurrences[token]++
+	ile.StemmedWordOccurrences = make(map[string]int32)
+
+	parser := tokenizer.New()
+	parser.AllowKeywordUnderscore()
+
+	// parse file content
```
Comment (fredxfred) [inline src/search/search_dictionary.go]: remove obvious comments like this

File: src/subcommands/lexical_search.go [hunk 2]
```diff
@@ -60,15 +61,59 @@ func LexicalSearch(args LexicalSearchArgs) ([]fileOccurrence, error) {
...
+	term := args.SearchTerm
+	stemmedTerm, err := snowball.Stem(term, "english", true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to stem word: %w", err)
+	}
+
+	stemmedResults := newStemmedSearchMap[stemmedTerm]
+	nonStemResults := newSearchMap[term]
+
+	// combine stemmed and non-stemmed occurrence numbers
```
Comment (fredxfred) [inline src/subcommands/lexical_search.go]: Make this comment more detailed, explain the "why" not the "what"

File: src/search/search_dictionary.go (hunk 1 above)
Comment (astradzhao) [reply src/search/search_dictionary.go]: i didnt think this was that obvious, but i changed it to be //parse & tokenize file content

File: src/subcommands/lexical_search.go (hunk 2 above)
Comment (astradzhao) [reply src/subcommands/lexical_search.go]: done

**Pull request merged**